ui/src/app/pages/admin/admin-institution-edit.tsx

```diff
@@ -0,0 +1,411 @@
+        institution.emailAddresses === institutionToEdit.emailAddresses &&
+        institution.emailDomains === institutionToEdit.emailDomains &&
+        institution.userInstructions === institutionToEdit.userInstructions &&
+        institution.organizationTypeOtherText === institutionToEdit.organizationTypeOtherText;
+  }
+
+
+  noEmptyRequiredFields() {
+    const {institution} = this.state;
+    let emailValid = true;
+    if (institution.duaTypeEnum) {
+      emailValid = institution.duaTypeEnum === DuaType.MASTER ?
+          institution.emailDomains !== undefined : institution.emailAddresses !== undefined;
+    }
+    return !emailValid || !institution.displayName || !institution.organizationTypeEnum ||
+      !institution.duaTypeEnum ||
+        (institution.organizationTypeEnum === OrganizationType.OTHER &&
+            !institution.organizationTypeOtherText);
+  }
+
+  // Disable save button if
+  // a) No fields were edited or if there are any errors
+  // b) email address/Domain are not valid
+  // c) Required fields are not empty
+  disableSave(errors) {
+    return this.noEmptyRequiredFields() || (errors && errors.displayName) || this.fieldsNotEdited()
+      || this.state.invalidEmailAddress || this.state.invalidEmailDomain;
+  }
+
+  async saveInstitution() {
+    const {institution} = this.state;
+    if (institution) {
+      this.setState({invalidEmailAddress: false});
+      if (institution.duaTypeEnum === DuaType.MASTER) {
+        institution.emailAddresses = [];
+      } else {
+        institution.emailDomains = [];
+      }
+      if (institution.organizationTypeEnum !== OrganizationType.OTHER) {
+        institution.organizationTypeOtherText = null;
+      }
+    }
+    if (this.props.urlParams.institutionId) {
+      await institutionApi().updateInstitution(this.props.urlParams.institutionId, institution)
+        .then(value => this.backNavigate())
+        .catch(reason => this.handleError(reason));
+    } else {
+      await institutionApi().createInstitution(institution)
+        .then(value => this.backNavigate())
+        .catch(reason => this.handleError(reason));
+    }
+  }
+
+  handleError(rejectReason) {
+    let errorMsg = 'Error while saving Institution. Please try again later';
+    if (rejectReason.status === 409) {
+      errorMsg  = 'Institution with Name ' + this.state.institution.displayName + ' already exist';
+    }
+    this.setState({apiErrorMsg: errorMsg, showApiError: true});
+  }
+  updateInstitutionRole(institutionRole) {
+    this.setState({showOtherInstitution: institutionRole === OrganizationType.OTHER});
+    this.setState(fp.set(['institution', 'organizationTypeEnum'], institutionRole));
+  }
+
+  backButton() {
+    if (!this.fieldsNotEdited()) {
+      this.setState({showBackButtonWarning: true});
+    } else {
+      this.backNavigate();
+    }
+  }
+
+  backNavigate() {
+    navigate(['admin/institution']);
+  }
+
+  isEmailAddressValid() {
```

Comment:
are these two checking for validity or presence?

Reply:
Better but I think we want to either rename this as "missing" or flip the sign here.